### PR TITLE
Fetch GitHub API to get GitHub Action Status

### PR DESCRIPTION
**Pull request by kolioOtSofia**
Closes: #46 

In order to test all the changes, put the following code in the workspace
```smalltalk
|repo|
repo := ATDDGitHubActionsRepository newForOwner: 'hpi-swa-teaching' repository: 'SWT-Demo'.
repo buildStatus 
```

### CODE REVIEW

File: packages/AutoTDD-Core.package/ATDDGitHubActionsRepository.class/instance/initialize.st [hunk 1]
```diff
@@ -0,0 +1,5 @@
+initialize-release
+initialize
+	
+	super initialize.
+	self parser: ATDDGitHubActionsParser new.
```
Comment (Benjamin-Frost) [inline packages/AutoTDD-Core.package/ATDDGitHubActionsRepository.class/instance/initialize.st:5]: nit: Remove dot

```suggestion
	self parser: ATDDGitHubActionsParser new
```

File: packages/AutoTDD-Core.package/ATDDGitHubActionsRepository.class/instance/client..st [hunk 2]
```diff
@@ -0,0 +1,4 @@
+accessing
+client: anATDDGitHubActionsClient
+	
+	client := anATDDGitHubActionsClient.
```
Comment (Benjamin-Frost) [inline packages/AutoTDD-Core.package/ATDDGitHubActionsRepository.class/instance/client..st:4]: nit: Remove dot

```suggestion
	client := anATDDGitHubActionsClient
```

File: packages/AutoTDD-Core.package/ATDDGitHubActionsRepository.class/instance/parser..st [hunk 3]
```diff
@@ -0,0 +1,4 @@
+accessing
+parser: anATTDGitHubActionsParser
+	
+	parser := anATTDGitHubActionsParser.
```
Comment (Benjamin-Frost) [inline packages/AutoTDD-Core.package/ATDDGitHubActionsRepository.class/instance/parser..st:4]: nit: Remove dot

```suggestion
	parser := anATTDGitHubActionsParser
```

**Review by Benjamin-Frost — CHANGES_REQUESTED** [review]

File: packages/AutoTDD-Core.package/ATDDGitHubActionsRepository.class/instance/client.st [hunk 4]
```diff
@@ -0,0 +1,3 @@
+accessing
+client
+	^ client ifNil: [ client := ATDDGitHubActionsClient new ]
```
Comment (SebastianWagner2) [inline packages/AutoTDD-Core.package/ATDDGitHubActionsRepository.class/instance/client.st:3]:
```suggestion
client

	^ client ifNil: [ client := ATDDGitHubActionsClient new ]
```

File: packages/AutoTDD-Core.package/ATDDGitHubActionsRepository.class/class/newForOwner.repository..st [hunk 5]
```diff
@@ -0,0 +1,7 @@
+class initialization
+newForOwner: anOwnerString repository: aRepositoryString
+	
+	^ self new
+			owner: anOwnerString;
+			repository: aRepositoryString;
+			yourself
```
Comment (SebastianWagner2) [inline packages/AutoTDD-Core.package/ATDDGitHubActionsRepository.class/class/newForOwner.repository..st]:
```suggestion remove one tab
	^ self new
		owner: anOwnerString;
		repository: aRepositoryString;
		yourself
```

**Review by SebastianWagner2 — APPROVED** [review]

File: packages/AutoTDD-Core.package/ATDDGitHubActionsRepository.class/instance/buildStatus.st [hunk 6]
```diff
@@ -0,0 +1,9 @@
+accessing
+buildStatus
+
+	| json parsedJson |
+	
+	json := self client getJsonForLatestWorkflowRunForRepository: owner, '/', repository, '/'.
```
Comment (SebastianWagner2) [inline packages/AutoTDD-Core.package/ATDDGitHubActionsRepository.class/instance/buildStatus.st:6]:
```suggestion
	json := self client getJsonForLatestWorkflowRunForRepository: owner , '/' , repository , '/'.
```

**Review by SebastianWagner2 — APPROVED** [review]

**Review by Benjamin-Frost — APPROVED** [review]

**Review by SebastianWagner2 — APPROVED** [review]

**Pull request merged**